sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/DLPDeidentifyText.java

```diff
@@ -0,0 +1,267 @@
+import org.apache.beam.sdk.values.PCollectionView;
+
+/**
+ * A {@link PTransform} connecting to Cloud DLP (https://cloud.google.com/dlp/docs/libraries) and
+ * deidentifying text according to provided settings. The transform supports both CSV formatted
+ * input data and unstructured input.
+ *
+ * <p>If the csvHeader property is set and a sideinput with CSV headers is added to the PTransform,
+ * csvDelimiter also should be set, else the results will be incorrect. If csvHeader is neither set
+ * nor passed as sideinput, input is assumed to be unstructured.
+ *
+ * <p>Either deidentifyTemplateName (String) or deidentifyConfig {@link DeidentifyConfig} need to be
+ * set. inspectTemplateName and inspectConfig ({@link InspectConfig} are optional.
+ *
+ * <p>Batch size defines how big are batches sent to DLP at once in bytes.
+ *
+ * <p>The transform consumes {@link KV} of {@link String}s (assumed to be filename as key and
+ * contents as value) and outputs {@link KV} of {@link String} (eg. filename) and {@link
+ * DeidentifyContentResponse}, which will contain {@link Table} of results for the user to consume.
+ */
+@Experimental
+@AutoValue
+public abstract class DLPDeidentifyText
+    extends PTransform<
+        PCollection<KV<String, String>>, PCollection<KV<String, DeidentifyContentResponse>>> {
+
+  public static final Integer DLP_PAYLOAD_LIMIT_BYTES = 524000;
+
+  /** @return Template name for data inspection. */
+  @Nullable
+  public abstract String inspectTemplateName();
+
+  /** @return Template name for data deidentification. */
+  @Nullable
+  public abstract String deidentifyTemplateName();
+
+  /**
+   * @return Configuration object for data inspection. If present, supersedes the template settings.
+   */
+  @Nullable
+  public abstract InspectConfig inspectConfig();
+
+  /** @return Configuration object for deidentification. If present, supersedes the template. */
+  @Nullable
+  public abstract DeidentifyConfig deidentifyConfig();
+
+  /** @return List of column names if the input KV value is a CSV formatted row. */
+  @Nullable
+  public abstract PCollectionView<List<String>> csvHeader();
+
+  /** @return Delimiter to be used when splitting values from input strings into columns. */
+  @Nullable
+  public abstract String csvColumnDelimiter();
+
+  /** @return Size of input elements batch to be sent to Cloud DLP service in one request. */
+  public abstract Integer batchSizeBytes();
+
+  /** @return ID of Google Cloud project to be used when deidentifying data. */
+  public abstract String projectId();
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    /** @param inspectTemplateName Template name for data inspection. */
+    public abstract Builder setInspectTemplateName(String inspectTemplateName);
+
+    /** @param csvHeader List of column names if the input KV value is a CSV formatted row. */
+    public abstract Builder setCsvHeader(PCollectionView<List<String>> csvHeader);
```

Comment:
Since this is just a collection of column names for the header, I don't think the prefix `csv` is relevant here. Below is my suggestion that drops the `csv`, adds the word 'column', and adds a comment that describes the relationship between this method and the `setColumnDelimiter`.

Note: I talk about the default header "value" here. It would make sense to mention this in `getHeaderColumns()` if my assumption is correct.

```suggestion
    /** 
    *  Sets the header column names corresponding to the delimited values of the input KV. If no header is set, 
    *  a single column name "value" will be used for the header. If specified, requires that a delimiter is 
    *  set using setColumnDelimiter.
    */
    public abstract Builder setHeaderColumns(PCollectionView<List<String>> header);
```